term-manager/impl/src/java/org/sakaiproject/acadtermmanage/logic/impl/AcademicSessionLogicImpl.java

```diff
@@ -0,0 +1,514 @@
+package org.sakaiproject.acadtermmanage.logic.impl;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+import org.sakaiproject.acadtermmanage.Constants;
+import org.sakaiproject.acadtermmanage.dao.AcademicSessionDao;
+import org.sakaiproject.acadtermmanage.exceptions.DuplicateKeyException;
+import org.sakaiproject.acadtermmanage.exceptions.NoSuchKeyException;
+import org.sakaiproject.acadtermmanage.logic.AcademicSessionLogic;
+import org.sakaiproject.acadtermmanage.logic.AcademicSessionSakaiProxy;
+import org.sakaiproject.acadtermmanage.model.Semester;
+import org.sakaiproject.authz.api.SecurityAdvisor;
+import org.sakaiproject.coursemanagement.api.AcademicSession;
+import org.sakaiproject.entity.api.ResourcePropertiesEdit;
+import org.sakaiproject.site.api.Site;
+
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import net.sf.ehcache.Cache;
+import net.sf.ehcache.Element;
+
+@Slf4j
+public class AcademicSessionLogicImpl implements AcademicSessionLogic {
+	
+	
+	/* 
+	 * I thought it would be nice to have non-admins be able to use this tool, so I added a permission
+	 * that could be checked by all methods which alter data.
+	 * But it turns out that the CM-API only allows admins to alter data, so using this permission 
+	 * (and then a security advisor which would grant the user all other necessary permissions)
+	 * instead of simply checking for admin status doesn't make sense right now.
+	 * So I'm disabling - but not removing - the custom-permission-based approach with this boolean switch.
+	 */
+	private static final boolean USE_PERMISSION=false;
+	private final SecurityAdvisor GOD_MODE = USE_PERMISSION?new GodMode():null;
```

Comment:
I feel this should be removed, only admins should be able to do this.
SecurityAdvisor is not a recommended practice

Reply:
Yeah, okay, I'll remove that code.